src/popup/components/InputAmount.tsx

```diff
@@ -44,67 +44,194 @@ export const InputAmount = ({
   min = 0,
   max,
   readOnly,
+  controls = false,
 }: Props) => {
+  const { assetScale } = walletAddress;
+  const step = 1 / 10 ** assetScale;
   const currencySymbol = getCurrencySymbol(walletAddress.assetCode);
 
-  const validateAmountOnChange = useThrottle(
-    (ev: React.ChangeEvent<HTMLInputElement>) => {
-      const input = ev.target;
-      const value = input.value;
+  const inputRef = React.useRef<HTMLInputElement>(null);
+
+  const formatAmount = React.useCallback(
+    (value: number) => formatNumber(value, assetScale),
+    [assetScale],
+  );
+
+  const handleValue = React.useCallback(
+    (value: string, skipSetValue = false) => {
```

Comment:
What are the exact cases when we want to skip the value?

Reply:
When user is typing, for example ”1", it'll get set formatted to ”1.00” with cursor moved to end on input field. So to avoid that, we skip that call. Inc/dec don't require such cursor, so ok (and even better) to format then.